src/builder/QuarkBuilder.sol

```diff
@@ -122,39 +217,44 @@ contract QuarkBuilder {
         Payment calldata payment,
         ChainAccounts[] calldata chainAccountsList
     ) external pure returns (BuilderResult memory) {
-        // TODO: Input validation: Check that arrays are equal length (e.g. chainIds and maxCosts in Payment)
-        AssetPositions[] paymentAssetAccounts;
-        // Get transfer and payment token
-        // TODO: implement filterAccounts
-        transferAssetAccounts = filterAccounts(assetSymbol, chainAccountsList);
+        if (payment.chainIds.length != payment.maxCosts.length) {
+            revert InvalidInput();
+        }
```

Comment:
This will be obviated by refactoring the max costs data into an array-of-structs.

However, raises the question: if a chain id is not included in maxCosts, does that mean we should not consider assets on that chain? It is easily possible for the `chainAccountsList` to contain chains that have no specified max costs.

So, is that an error?
Or is that informative? Ie., if maxCosts is not empty, are those the only chains we should consider using?

Reply:
Yeah, this was removed after I added the `MaxPaymentCost` struct.

Hmm, interesting question. I don't think we should require the client to supply max costs for every chain. So, we could either 1) not construct Quark Operations on chains where max cost is unset or 2) choose some default value if max cost is unset. 

I do prefer 1, since there really isn't a good default value that works for all Quark Operations and payment tokens for 2. It's also a bit scary to set a value for it that's not specified by the client.

Reply:
Agreed, I think if no max is provided, we should assume a max of 0.

Reply:
With the edge case that, if `maxCosts` is empty, we must still return a result that can be provided to simulate.

We may expect the client to explicitly pass in some large numbers for `maxCosts`, alternatively. :shrug: 

It's also perhaps a situation that will overlap with our handling of "maximum" or "unlimited" in the case of "transfer max" -- if we choose a sigil like `uint256(-1)` to represent "max" or "no limit," then maybe that's what we should use to explicitly construct simulations ignoring cost limits...